pkg/fqdn/namemanager/gc_cache.go

```
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package namemanager
+
+import (
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// DefaultGCCacheSize is the default size of the GC cache
+const DefaultGCCacheSize = 10000
+
+// GCCache tracks recently garbage-collected FQDN entries.
+type GCCache struct {
+	cache *lru.Cache[string, time.Time]
+	mu    sync.RWMutex
+}
+
+// GCCacheEntry represents a garbage-collected FQDN entry.
+type GCCacheEntry struct {
+	FQDN                  string    `json:"fqdn"`
+	GarbageCollectionTime time.Time `json:"garbage-collection-time"`
+}
+
+// NewGCCache creates a new GCCache with the specified size.
+func NewGCCache(size int) *GCCache {
```

Not a big fan of panic on exported apis. Could we return the error and let the call site decide?

---

tagging @gandro for a final decision here. I had initially returned the error, but had switched to a panic under @gandro's suggestion. 

Here was the original comment
> Let's just panic here if lru.New returns an error (this only happens if size is less than zero, which is impossible here, and basically always a programmer or configuration mistake)

---

I still strongly feel a panic is the right thing to do if the users passes in a negative capacity. It's also what the Go does if you do something like `make([]string, 0, negativeInt)`.